asserts/user_test.go

```diff
@@ -0,0 +1,145 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package asserts_test
+
+import (
+	"strings"
+	"time"
+
+	"github.com/snapcore/snapd/asserts"
+	. "gopkg.in/check.v1"
+)
+
+var (
+	_ = Suite(&systemUserSuite{})
+)
+
+type systemUserSuite struct {
+}
+
+const systemUserExample = "type: system-user\n" +
+	"authority-id: canonical\n" +
+	"brand-id: canonical\n" +
+	"email: foo@example.com\n" +
+	"series:\n" +
+	"  - 16\n" +
+	"models:\n" +
+	"  - frobinator\n" +
+	"name: Nice Guy\n" +
+	"username: guy\n" +
+	"password: $6$salt$hash\n" +
+	"ssh-keys:\n" +
+	"  - ssh-rsa AAAABcdefg\n" +
+	"since: 1092-11-01T22:08:41+00:00\n" +
+	"until: 2092-11-01T22:08:41+00:00\n" +
+	"body-length: 0\n" +
+	"sign-key-sha3-384: Jv8_JiHiIzJVcO9M55pPdqSDWUvuhfDIBJUS-3VW7F_idjix7Ffn5qMxB21ZQuij" +
+	"\n\n" +
+	"AXNpZw=="
+
+func (s *systemUserSuite) TestDecodeOK(c *C) {
+	a, err := asserts.Decode([]byte(systemUserExample))
+	c.Assert(err, IsNil)
+	c.Check(a.Type(), Equals, asserts.SystemUserType)
+	systemUser := a.(*asserts.SystemUser)
+	c.Check(systemUser.BrandID(), Equals, "canonical")
+	c.Check(systemUser.EMail(), Equals, "foo@example.com")
+	c.Check(systemUser.Series(), DeepEquals, []string{"16"})
+	c.Check(systemUser.Models(), DeepEquals, []string{"frobinator"})
+	c.Check(systemUser.Name(), Equals, "Nice Guy")
+	c.Check(systemUser.Username(), Equals, "guy")
+	c.Check(systemUser.Password(), Equals, "$6$salt$hash")
+	c.Check(systemUser.SSHKeys(), DeepEquals, []string{"ssh-rsa AAAABcdefg"})
+	since, err := time.Parse(time.RFC3339, "1092-11-01T22:08:41+00:00")
+	c.Assert(err, IsNil)
+	c.Check(systemUser.Since(), DeepEquals, since)
+
+	until, err := time.Parse(time.RFC3339, "2092-11-01T22:08:41+00:00")
+	c.Assert(err, IsNil)
+	c.Check(systemUser.Until(), DeepEquals, until)
+}
+
+func (s *systemUserSuite) TestDecodePasswd(c *C) {
+	validTests := []struct{ original, valid string }{
+		{"password: $6$salt$hash\n", "password: $6$rounds=9999$salt$hash\n"},
+	}
+	for _, test := range validTests {
+		valid := strings.Replace(systemUserExample, test.original, test.valid, 1)
+		_, err := asserts.Decode([]byte(valid))
+		c.Check(err, IsNil)
+	}
+}
+
+func (s *systemUserSuite) TestValidAt(c *C) {
```

Comment:
Thanks, I added a test for the since==until now. The other thing that is tested in TestPublicKeyIsValid is a missing until field but we do not allow this with this assertion.